Add spec file to machine-driver-libvirt

cfergeau opened the pull request:
This adds a `make spec` target which generates a spec file for use on Red Hat infrastructure.
I did not test this .spec file against fedora, but hopefully it works there as well.
This introduces a gomod2rpmreqs helper to convert the list of go module used by
machine-driver-libvirt to virtual rpm Requires, for better tracking of what
code machine-driver-libvirt is vendoring.

cfergeau committed:
build: Add spec file

This spec file can then be used to trigger a build on
a RHEL8 system. It uses the 'vendor' directory (ie does not expect the
bundled dependencies to be split in separate packages), but uses the
gomod2rpmdeps tool to document which go modules are bundled.

gbraad commented on cmd/gomod2rpmdeps/gomod2rpmdeps.go:
```diff
@@ -162,6 +162,12 @@ func pseudoVersionToRpmVersion(pseudoVersion string) (string, error) {
 		}
 
 		return fmt.Sprintf("%s+git%s.%s", strings.TrimPrefix(substrings[0], "v"), substrings[1], substrings[2]), nil
+		// FIXME: these are not parsed properly. '0.' and 'alpha.' are documented here https://golang.org/ref/mod#pseudo-versions
```
can't this be part of the previous commit? especially as it addresses a 'crtitical' shortcoming in the initial/current implementation

cfergeau replied:
Sure, I can squash, wasn't sure whether to make this more obvious through a separate commit, to remove, to squash.

cfergeau commented:
I've moved the gomod2rpmdeps helper to https://github.com/cfergeau/gomod2rpmdeps , I'll need to update this PR.

cfergeau committed:
Add test-rpmbuild Makefile target

This adds a Containerfile.rpmbuild file in order to test
the new spec file in a centos container.

cfergeau committed:
circleci: Update to newer ubuntu image

The image we use is an older ubuntu, with an older golang.
Better to update to a newer image.

guillaumerose commented:
The code of gomod2rpmdeps is still here. Is it ok?

cfergeau commented:
I haven't updated this PR yet, I'll update it this week.

cfergeau commented:
Not sure why CI fails, haven't been able to reproduce.

guillaumerose commented:
Works also on my machine :thinking:

cfergeau commented:
This can be reproduced with `GOPATH=~/go/bin:~/.local/bin make`
